docs/proposals/dockerfile_component_detection.md

```diff
@@ -0,0 +1,58 @@
+# Enhance port detection for dockerfiles that are not in a component.
+
+## Related GitHub Issues
+* https://github.com/redhat-developer/alizer/issues/235
+
+## Background
+
+What is a component?
+
+Components are defined in the `languages-customization.yml` as a language that has specfic `configuration_files`.
+
+Example of a Go component:
+
+```yaml
+Go:
+  exclude_folders:
+    - "vendor"
+  configuration_files:
+    - "go.mod"
+  component: true
+```
+
+What happens when the repo being scanned has component(s) found but also contains a dockerfile one level outside of the repo?
+
+```shell
+.repo-being-detected-by-alizer
+├── component/
+│   └── dockerfile
+└── dockerfile
+```
+
+The dockerfile in the root level would not be used during analysis because it does not have a component definition in the `languages-customization.yml`
+
+## Potential Solution
+
+Introduce a new component field called `orphan_component` and add it to the `Dockerfile` language:
+
+```yaml
+Dockerfile:
+  configuration_files:
+    - "[Dd]ockerfile"
+  orphan_component: true
+```
+
+and create a component enricher:
+
+```go
+func (d DockerEnricher) DoEnrichComponent(component *model.Component, settings model.DetectionSettings, ctx *context.Context) {
+    // get ports from the dockerfile using our existing dockerfile strategy
+}
+```
+
+Using the `orphan_component` field, we can separate out a dockerfile as a component from what we currently define as a component. This can also be used for any future components that are similar to a dockerfile.
+
+## Potential Problems
```

Comment:
It is good that we identify the potential problems, however, we haven't mentioned the action for addressing them as part of the proposal. If we think these are problems that need to be addressed but are out of the scope for this proposal, should we create an issue to track those and add a link to the issues here?

Reply:
Yes agreed! We can change the original issue into an epic, and then also add all related issues to it too as well as linking the issues in the proposal